hack/build-antrea-windows-all.sh

```diff
@@ -29,6 +29,7 @@ typically used in CI to build the image with the latest version of all dependenc
 account changes to all Dockerfiles.
         --pull                  Always attempt to pull a newer version of the base images.
         --push-base-images      Push built images to the registry. Only base images will be pushed.
+        --skip-base-images      Skip building base images if they exists. Only build them when image pulling fails.
```

Comment:
I think the point of the script is to rebuild all images with the latest version of all dependencies, as seen from the usage description and the fact that it is used in docker_update_base_windows.yml.
And the skipped base images don't mean the same images as the ones meant by `--push-base-images`.

Reply:
For the skip case, now it is only used for CI pipeline. The base image we use in skip case is the latest image pushed by "--push-base-images". Therefore if any error occurs in the base image the CI pipeline can detect it as soon as possible. In current pipeline CI will always rebuild base image, which means it may different from the image pushed by docker_update_base_windows.yml. 

Last time we found harbor base image error but CI still works normally. I think we should update the way CI build image to align with the base image pushed by docker_update_base_windows.yml.

Reply:
> Build the antrea/antrea-windows image, as well as all the base images in the build chain. This is
typically used in CI to build the image with the latest version of all dependencies, taking into
account changes to all Dockerfiles.

I meant the script's usage description says the script it's meant to rebuild all the images with latest version of all dependecies. Adding this flag makes it no longer true. I don't get how changing the script in this way would avoid inconsistent images being used and I think it would make the issue happen again as CI test would rebuild the base image when it fails to pull the one used by "build-windows" job.

If I understand correctly, both "build-windows" job and CI test should only use the base images that are already present in dockerhub. When dependencies' versions change, we should schedule "Manually update antrea/base-windows Docker image" job to build the new image. Automatically rebuilding missing base images may encounter same issue as before and doesn't save the effort to manually schedule the job anyway.

I guess @antoninbas's comment in https://github.com/antrea-io/antrea/pull/4374#discussion_r1014273843 meant same thing, not moving building base-image from Makefile to script.